src/Bridges/Variable/bridge.jl

```diff
@@ -0,0 +1,157 @@
+
+Subtype of [`MathOptInterface.Bridges.AbstractBridge`](@ref) for variable
+bridges.
+"""
+abstract type AbstractBridge <: MOIB.AbstractBridge end
+
+"""
+    IndexInVector
+
+Index of variable in vector of variables.
+"""
+struct IndexInVector
+    value::Int
+end
+
+"""
+    bridge_constrained_variable(BT::Type{<:AbstractBridge}, model::MOI.ModelLike,
+                                set::MOI.AbstractSet)
+
+Bridge the constrained variable in `set` using bridge `BT` to `model` and returns
+a bridge object of type `BT`. The bridge type `BT` should be a concrete type,
+that is, all the type parameters of the bridge should be set. Use
+[`concrete_bridge_type`](@ref) to obtain a concrete type for given set types.
+"""
+function bridge_constrained_variable end
+
+"""
+    function MOI.get(model::MOI.ModelLike, attr::MOI.AbstractVariableAttribute,
+                     bridge::AbstractBridge)
+
+Return the value of the attribute `attr` of the model `model` for the
+variable bridged by `bridge`.
+"""
+function MOI.get(::MOI.ModelLike, attr::MOI.AbstractVariableAttribute,
+                 bridge::AbstractBridge)
+    throw(ArgumentError("Variable bridge of type `$(typeof(bridge))` does not support accessing the attribute `$attr`."))
+end
+
+"""
+    function MOI.get(model::MOI.ModelLike, attr::MOI.AbstractVariableAttribute,
+                     bridge::AbstractBridge, i::IndexInVector)
+
+Return the value of the attribute `attr` of the model `model` for the
+variable at index `i` in the vector of variables bridged by `bridge`.
+"""
+function MOI.get(::MOI.ModelLike, attr::MOI.AbstractVariableAttribute,
+                 bridge::AbstractBridge, ::IndexInVector)
+    throw(ArgumentError("Variable bridge of type `$(typeof(bridge))` does not support accessing the attribute `$attr`."))
+end
+
+"""
+    supports_constrained_variable(::Type{<:AbstractBridge},
+                                   ::Type{<:MOI.AbstractSet})::Bool
+
+Return a `Bool` indicating whether the bridges of type `BT` support bridging
+constrained variables in `S`.
+"""
+function supports_constrained_variable(::Type{<:AbstractBridge},
+                                        ::Type{<:MOI.AbstractSet})
+    return false
+end
+
+"""
+    added_constrained_variable_types(BT::Type{<:MOI.Bridges.Variable.AbstractBridge},
+                                     S::Type{<:MOI.AbstractSet})
+
+Return a list of the types of constrained variables that bridges of type `BT`
+add for bridging constrained variabled in `S`. This fallbacks to
+`added_constrained_variable_types(concrete_bridge_type(BT, S))`
+so bridges should not implement this.
+
+## Examples
+
+As a variable in [`MathOptInterface.GreaterThan`](@ref) is bridged into
+variables in [`MathOptInterface.Nonnegatives`](@ref) by the
+[`VectorizeBridge`](@ref):
+```jldoctest
+BT = MOI.Bridges.Variable.VectorizeBridge{Float64}
+S = MOI.GreaterThan{Float64}
+MOI.Bridges.added_constrained_variable_types(BT, S)
+
+# output
+
+[(MOI.Nonnegatives,)]
```

Comment:
These examples are very helpful. However, I don't understand how the doctests are passing given that the bridges are not implemented yet.

Reply:
The function docstring is not included in the doc, it is included in the `bl/variable_bridge` branch